java/query/query-runner/src/main/java/sleeper/query/runner/tracker/DynamoDBQueryTracker.java

```diff
@@ -219,10 +216,13 @@ private void updateStateOfParent(DynamoDBQueryTrackerEntry leafQueryEntry) {
         QueryState parentState = getParentState(children);
 
         if (parentState != null) {
-            long totalRecordCount = children.stream().mapToLong(query ->
-                    query.getRecordCount() != null ? query.getRecordCount() : 0).sum();
+            long totalRecordCount = children.stream()
+                    .mapToLong(query -> query.getRecordCount() != null ? query.getRecordCount() : 0).sum();
             LOGGER.info("Updating state of parent to {}", parentState);
             updateState(leafQueryEntry.updateParent(parentState, totalRecordCount));
+        } else {
+            LOGGER.info("Not updating state of parent query, {} leaf queries are still in progress",
+                    children.stream().filter(child -> child.getLastKnownState() != QueryState.COMPLETED).count());
```

Comment:
This isn't quite accurate, this will count queries as in progress when they actually failed. Maybe this logging should be done in getParentState to make sure the logic matches?